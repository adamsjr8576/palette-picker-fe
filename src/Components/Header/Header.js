import React from 'react';
import { connect } from 'react-redux';
import './Header.scss';

const Header = ({ currentPalette }) => {
  let borderColor;
  if (currentPalette.length) {
    borderColor = currentPalette[0].color;
  }
  return (
    <header className="main-header" style={{borderBottom: `12px groove ${borderColor}`}}>
      <h1 className="header-title">Palette Picker</h1>
    </header>
  )
}

export const mapStateToProps = state => ({
  currentPalette: state.currentPalette
});

export default connect(mapStateToProps)(Header);
