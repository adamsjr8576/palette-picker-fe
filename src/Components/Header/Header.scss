.main-header {
  height: 100px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.header-title {
  font-size: 2.5em;
  margin: 0;
}
