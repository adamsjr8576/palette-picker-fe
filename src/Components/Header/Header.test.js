import React from 'react';
import { Header, mapStateToProps } from './Header'
import { shallow } from 'enzyme';

describe('Header', () => {

  let wrapper, mockCurrentPalette;

  describe('Header Component', () => {

    it('should match the snapshot when currentPalette is an empty array', () => {
      mockCurrentPalette = [];
      wrapper = shallow(<Header currentPalette={mockCurrentPalette}/>)
      expect(wrapper).toMatchSnapshot()
    });

    it('should match the snapshot when currentPalette is an array of colors', () => {
      mockCurrentPalette = [
        {locked: false, color: '#11111'},
        {locked: false, color: '#11111'},
        {locked: false, color: '#11111'},
        {locked: false, color: '#11111'},
        {locked: false, color: '#11111'}];
      wrapper = shallow(<Header currentPalette={mockCurrentPalette}/>)
      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with the currentPalette data', () => {
      const mockState = {
        currentPalette: [
          {locked: false, color: '#444343'},
          {locked: false, color: '#6D6D6D'},
          {locked: false, color: '#9B9B9B'},
          {locked: false, color: '#C2C2C2'},
          {locked: false, color: '#DCDCDC'}],
        projects: []
      };
      const expected = {
        currentPalette: [
          {locked: false, color: '#444343'},
          {locked: false, color: '#6D6D6D'},
          {locked: false, color: '#9B9B9B'},
          {locked: false, color: '#C2C2C2'},
          {locked: false, color: '#DCDCDC'}]
      };

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});
