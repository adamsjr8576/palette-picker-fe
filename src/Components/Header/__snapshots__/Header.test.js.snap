// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Component should match the snapshot when currentPalette is an array of colors 1`] = `
<header
  className="main-header"
  style={
    Object {
      "borderBottom": "12px groove #11111",
    }
  }
>
  <h1
    className="header-title"
  >
    Palette Picker
  </h1>
</header>
`;

exports[`Header Header Component should match the snapshot when currentPalette is an empty array 1`] = `
<header
  className="main-header"
  style={
    Object {
      "borderBottom": "12px groove undefined",
    }
  }
>
  <h1
    className="header-title"
  >
    Palette Picker
  </h1>
</header>
`;
